README
Proyecto del curso Plataformas Abiertas
LunarLander v1
2018, Dic 8
Leonardo Agüero Villagra - leoagv88@gmail.com
María Jesús Aguilar Esquivel - mariajesus1399@gmail.com
Daniel Campos Salas - daniel.campos138@gmail.com
License WTFPL4.
Este es el juego Lunar Lander.
El videojuego es una simulación del acontecimiento histórico del alunizaje del 
hombre realizado por los Estados Unidos el 21 de Julio de 1969.
El videojuego consiste en aterrizar de manera correcta una nave espacial, 
sin chocar en el proceso, para esto se tiene combustible y se debe aterrizar a
una velocidad baja para así salvar a la tripulación. 


CHANGELOG
(07/12/18)[1:00 p.m]: se introdujo el main principal
(07/12/18)[11:21 p.m]: listo makefile v1.
(08/12/18)[12:00 p.m]: listo diseño de fondo del menú y botones del menú.
(08/12/18)[2:30 p.m]: listo juego funcionando con fondo del menú principal.
(08/12/18)[3:40 p.m]: música del juego funciona.
(09/12/18)[11:30 p.m]: Menú responde al mouse y botón salir funciona.
(10/12/18)[4:32 p.m]: Tecla empezar inicia la pantalla de juego.


main.c
/* Main.c
El Main del juego
Autores:
Leonardo Agüero Villagra
-
-
*/

#include <allegro5/allegro.h>
#include <allegro5/allegro_audio.h>
#include <allegro5/allegro_acodec.h>
#include <allegro5/allegro_image.h>
#include "juegoActivo.h"

#include <stdlib.h>
#include <stdio.h>
#include <math.h>


int main(int argc, char **argv)
{

    al_init_font_addon();
    al_init_ttf_addon();
    int running;

    ALLEGRO_DISPLAY * display;
    ALLEGRO_SAMPLE *sample = NULL;
    ALLEGRO_EVENT_QUEUE * queue;
    ALLEGRO_TIMER * timer;

    //para imagenes
    ALLEGRO_BITMAP * menuBackground = NULL;
    ALLEGRO_BITMAP * onClickEmpezar = NULL;
    ALLEGRO_BITMAP * onClickSalir = NULL; 
    ALLEGRO_BITMAP * empezarbackground = NULL;
    ALLEGRO_BITMAP * nave = NULL;

    al_init();

    if(!al_init()){
        fprintf(stderr, "failed to initialize allegro!\n");
        return -1;
    }
    if(!al_install_audio()){
        fprintf(stderr, "failed to initialize audio!\n");
        return -1;
    }

    if(!al_init_acodec_addon()){
        fprintf(stderr, "failed to initialize audio codecs!\n");
        return -1;
    }

    if (!al_reserve_samples(1)){
        fprintf(stderr, "failed to reserve samples!\n");
        return -1;
    }

    sample = al_load_sample("resources/sound/musica.ogg");

    if (!sample){
        printf( "Audio clip sample not loaded!\n" );
        return -1;
    }

    display = al_create_display(800, 530);

    if(!display){
        fprintf(stderr, "failed to create display!\n");
        return -1;
    }
 
    queue = al_create_event_queue();
    timer = al_create_timer(1.0 / 60.0);

    al_init_image_addon();
    al_install_keyboard();
    
    //para el mouse
    al_install_mouse();
    ALLEGRO_MOUSE_STATE mousepos;
    al_get_mouse_state(&mousepos);


    al_register_event_source(queue, al_get_keyboard_event_source());
    al_register_event_source(queue, al_get_mouse_event_source());
    al_register_event_source(queue, al_get_display_event_source(display));
    al_register_event_source(queue, al_get_timer_event_source(timer));

    menuBackground = al_load_bitmap("resources/background/menuBackground.png");
    onClickEmpezar = al_load_bitmap("resources/background/onClickEmpezar.png");
    onClickSalir = al_load_bitmap("resources/background/onClickSalir.png");
    empezarbackground = al_load_bitmap("resources/background/empezarbackground.png");
    nave = al_load_bitmap("resources/background/nave.png");
    

    running = 1;
    float mouseposx = 0;
    float mouseposy = 0;
    al_start_timer(timer);

    while(running == 1) {
        ALLEGRO_EVENT event;
        ALLEGRO_KEYBOARD_STATE keystate;
        al_wait_for_event(queue, &event);

        //si se toca la x de salida sale
        if(event.type == ALLEGRO_EVENT_DISPLAY_CLOSE){
            running = 0;
        }

        //sale del juego si se toca boton salir
        if((event.type == ALLEGRO_EVENT_MOUSE_BUTTON_DOWN) && (mouseposx > 127 && mouseposx < 670 && mouseposy > 290 && mouseposy < 380) ){
            running = 0;
        }

        //empieza el juego
        if((event.type == ALLEGRO_EVENT_MOUSE_BUTTON_DOWN) && (mouseposx > 127 && mouseposx < 670 && mouseposy > 175 && mouseposy < 264)){
            int onEmpezar;
            juego_activo(&queue, &event, &onEmpezar, &nave, &empezarbackground, &keystate);
        }

        //posición del mouse
        if(event.type == ALLEGRO_EVENT_MOUSE_AXES){
            mouseposx = event.mouse.x;
            mouseposy = event.mouse.y;
        }

        //Pone la música.
        al_play_sample(sample, 1.0, 0.0,1.0,ALLEGRO_PLAYMODE_LOOP,NULL);

        if(event.type == ALLEGRO_EVENT_TIMER){

            if(mouseposx > 127 && mouseposx < 670 && mouseposy > 175 && mouseposy < 264){
                al_draw_bitmap(onClickEmpezar, 0, 0, 0);
            }else if(mouseposx > 127 && mouseposx < 670 && mouseposy > 290 && mouseposy < 380){
                al_draw_bitmap(onClickSalir, 0, 0, 0);                
            }else{
                al_draw_bitmap(menuBackground, 0, 0, 0);
            }

            al_flip_display();

        }
    }

    //Libera memorias
    al_destroy_display(display);
    al_destroy_sample(sample);
    al_destroy_event_queue(queue);
    al_destroy_timer(timer);
    al_destroy_bitmap(menuBackground);
    al_destroy_bitmap(onClickEmpezar);
    al_destroy_bitmap(onClickSalir);
    al_destroy_bitmap(empezarbackground);
    al_destroy_bitmap(nave);
    al_uninstall_keyboard();
    al_uninstall_mouse();

    return 0;
}

src/ganar.c
#include "juegoActivo.h"

void ganar(ALLEGRO_BITMAP ** empezarbackground, ALLEGRO_EVENT_QUEUE ** queue, ALLEGRO_EVENT * event, ALLEGRO_KEYBOARD_STATE* keystate, int * fuel, int * score)
{
    int onGanar;
    ALLEGRO_FONT * menu;
    ALLEGRO_FONT * combustible;
    ALLEGRO_FONT * puntaje;

    menu = al_load_font("resources/fonts/AbyssinicaSIL-R.ttf", 36, 0);
    combustible = al_load_font("resources/fonts/AbyssinicaSIL-R.ttf", 36, 0);
    puntaje = al_load_font("resources/fonts/AbyssinicaSIL-R.ttf", 36, 0);

    *score = (*score) * (*fuel);
    onGanar = 2;

    while(onGanar == 2){
        al_wait_for_event(*queue, event);
        al_get_keyboard_state(keystate);
        al_draw_bitmap(*empezarbackground, 0 , 0, 0);
        
        if(event->type == ALLEGRO_EVENT_DISPLAY_CLOSE){
            onGanar = 1;
        }

        if(al_key_down(keystate, ALLEGRO_KEY_ENTER)){
            onGanar = 0;
        }

        if(event->type == ALLEGRO_EVENT_TIMER){
            al_draw_text(menu, al_map_rgb(255,255,255), 400, 160, ALLEGRO_ALIGN_CENTER, "Has ganado el juego");
            al_draw_textf(combustible, al_map_rgb(255,255,255), 400, 210, ALLEGRO_ALIGN_CENTER, "Combustible final: %d", *fuel);
            al_draw_textf(puntaje, al_map_rgb(255,255,255), 400, 260, ALLEGRO_ALIGN_CENTER, "Puntaje: %d", *score);
            al_draw_text(menu, al_map_rgb(255,255,255), 400, 310, ALLEGRO_ALIGN_CENTER, "Presiona ENTER para ir al menú principal");
            al_flip_display();
        }
    }
    al_destroy_font(puntaje);
    al_destroy_font(combustible);
    al_destroy_font(menu);
}

src/pausa.c
#include "juegoActivo.h"

int pausa(ALLEGRO_BITMAP ** empezarbackground, ALLEGRO_EVENT_QUEUE ** queue, ALLEGRO_EVENT * event, ALLEGRO_KEYBOARD_STATE* keystate)
{
    int onPausa = 2;
    ALLEGRO_FONT * menu;
    ALLEGRO_FONT * reanudar;

    menu = al_load_font("resources/fonts/AbyssinicaSIL-R.ttf", 36, 0);
    reanudar = al_load_font("resources/fonts/AbyssinicaSIL-R.ttf", 36, 0);
 
    while(onPausa == 2){
        al_wait_for_event(*queue, event);
        al_get_keyboard_state(keystate);
        al_draw_bitmap(*empezarbackground, 0 , 0, 0);
        
        if(event->type == ALLEGRO_EVENT_DISPLAY_CLOSE){
            onPausa = 1;
        }

        if(al_key_down(keystate, ALLEGRO_KEY_SPACE)){
            onPausa = 1;
        }

        if(al_key_down(keystate, ALLEGRO_KEY_ENTER)){
            onPausa = 0;
        }

        if(event->type == ALLEGRO_EVENT_TIMER){
            al_draw_text(menu, al_map_rgb(255,255,255), 400, 295, ALLEGRO_ALIGN_CENTER, "Presione ENTER para ir al menú principal");
            al_draw_text(reanudar, al_map_rgb(255,255,255), 400, 235, ALLEGRO_ALIGN_CENTER, "Presione SPACE para reanudar");
            al_flip_display();
        }

    }

    al_destroy_font(menu);
    al_destroy_font(reanudar);

    return onPausa;
}

src/perder.c
#include "juegoActivo.h"

void perder(ALLEGRO_BITMAP ** empezarbackground, ALLEGRO_EVENT_QUEUE ** queue, ALLEGRO_EVENT * event, ALLEGRO_KEYBOARD_STATE* keystate)
{
    int onPerder = 2;
    ALLEGRO_FONT * menu;

    menu = al_load_font("resources/fonts/AbyssinicaSIL-R.ttf", 36, 0);

    while(onPerder == 2){
        al_wait_for_event(*queue, event);
        al_get_keyboard_state(keystate);
        al_draw_bitmap(*empezarbackground, 0 , 0, 0);
        
        if(event->type == ALLEGRO_EVENT_DISPLAY_CLOSE){
            onPerder = 1;
        }

        if(al_key_down(keystate, ALLEGRO_KEY_ENTER)){
            onPerder = 0;
        }

        if(event->type == ALLEGRO_EVENT_TIMER){
            al_draw_text(menu, al_map_rgb(255,255,255), 400, 215, ALLEGRO_ALIGN_CENTER, "Has perdido");
            al_draw_text(menu, al_map_rgb(255,255,255), 400, 265, ALLEGRO_ALIGN_CENTER, "Presiona ENTER para ir al menú principal");
            al_flip_display();
        }

    }
    al_destroy_font(menu);
}

LunarLander/src/juegoActivo.c
#include "juegoActivo.h"


float velocidad(float * v_ini, float * t, float g)
{
    float velocidad;
    velocidad = (0.8)*(*v_ini + g * (*t));
    return velocidad;
}

float gravedad(float g, float * v_ini, float * cont)
{
    float caida; //es una posición y
    caida = (*v_ini) * (*cont) + (g * ((*cont)*(*cont)));
    return caida;
}

void juego_activo(ALLEGRO_EVENT_QUEUE ** queue, ALLEGRO_EVENT * event, int * onEmpezar, ALLEGRO_BITMAP ** nave, ALLEGRO_BITMAP ** empezarbackground, ALLEGRO_KEYBOARD_STATE * keystate)
{
    
    ALLEGRO_TIMER * gametime;
    ALLEGRO_FONT * displayFuel;
    ALLEGRO_FONT * displayVelocity;
    ALLEGRO_BITMAP * plataforma1 = NULL;

    displayFuel = al_load_font("resources/fonts/AbyssinicaSIL-R.ttf", 12, 0);
    displayVelocity = al_load_font("resources/fonts/AbyssinicaSIL-R.ttf", 12, 0);

    plataforma1 = al_load_bitmap("resources/background/plataforma1.png");

    gametime = al_create_timer(1.0 / 60.0);


    *onEmpezar = 1;
    float x = 10.0;
    int fuel = 8000;
    float y = 30.0;
    float v = 0;
    float cont = 0.0;
    float cont2 = 0.0;
    int score = 0;

    al_start_timer(gametime);
    while(*onEmpezar == 1){
        al_wait_for_event(*queue, event);
        al_get_keyboard_state(keystate);
        al_draw_bitmap(*empezarbackground, 0 , 0, 0);

        if(al_key_down(keystate, ALLEGRO_KEY_RIGHT)){
            x += 0.8;
            fuel -= 2.0;
        }
        if(al_key_down(keystate, ALLEGRO_KEY_LEFT)){
            x -= 0.8;
            fuel -= 2.0;
        }

        if(!(al_key_down(keystate, ALLEGRO_KEY_SPACE))){
            cont = cont + (1.0/60.0);
            v = velocidad(&v, &cont, 0.2);
            y = y + gravedad(0.1, &v, &cont);
            cont2 = (0.4)*cont;
        }else if((al_key_down(keystate, ALLEGRO_KEY_SPACE))){
            cont2 = cont2 + (1.0/60.0);
            v = velocidad(&v, &cont2, -0.3);
            y = y + gravedad(-0.15, &v, &cont2);
            cont = (0.4)*cont2;
            fuel -= 6;
        }

        if(fuel < 0){
            *onEmpezar = 0;
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }

        if(event->type == ALLEGRO_EVENT_DISPLAY_CLOSE){
            *onEmpezar = 0;
        }

        if(al_key_down(keystate, ALLEGRO_KEY_P)){
            *onEmpezar = pausa(&*empezarbackground, &*queue, &*event, &*keystate);
        }
    
        if (y > 305 && x < 93 && x > 0){
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 407 && x > 93 && x < 112) {
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 407 && x > 173 && x < 184) {
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 495 && x > 184 && x < 233){
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 241 && x > 233 && x < 279){
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 241 && x > 337 && x < 353){
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > (2.2232*x)-530 && x > 353 && x < 468){
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 494 && x > 468 && x < 498){
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 397 && x > 498 && x < 543){
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 313 && x > 597 && x < 678){
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 503 && x > 678 && x < 714){
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 503 && x > 774 && x < 800){
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }

        if (y > 402 && x > 112 && x < 173 && v < 0.6){
            score= score + 100;
            *onEmpezar = 0;/*GANA*/
            juego_activo2(&*queue, &*event, &*onEmpezar, &*nave, &*empezarbackground, &*keystate, &fuel, &score);
        }
        if (y > 238 && x > 279 && x < 337 && v < 0.6){
            score= score + 200;
            *onEmpezar = 0;/*GANA*/
            juego_activo2(&*queue, &*event, &*onEmpezar, &*nave, &*empezarbackground, &*keystate, &fuel, &score);
        }
        if (y > 392 && x > 543 && x < 597 && v < 0.6){
            score= score + 300;
            *onEmpezar = 0;/*GANA*/
            juego_activo2(&*queue, &*event, &*onEmpezar, &*nave, &*empezarbackground, &*keystate, &fuel, &score);

        }
        if (y > 495 && x > 714 && x < 774 && v < 0.6){
            score= score + 400;
            *onEmpezar = 0;/*GANA*/
            juego_activo2(&*queue, &*event, &*onEmpezar, &*nave, &*empezarbackground, &*keystate, &fuel, &score);
        }

        //------------Contrario a ganar------------------------------------

        if (y > 402 && x > 112 && x < 173 && v >= 0.6){
            *onEmpezar = 0;
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 238 && x > 279 && x < 337 && v >= 0.6){
            *onEmpezar = 0;
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 392 && x > 543 && x < 597 && v >= 0.6){
            *onEmpezar = 0;
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 495 && x > 714 && x < 774 && v >= 0.6){
            *onEmpezar = 0;
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }

        //si sale de pantalla de juego
        if(x < 0 || x > 765){
            *onEmpezar = 0;
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }


        if(event->type == ALLEGRO_EVENT_TIMER){
            al_draw_text(displayFuel, al_map_rgb(255,255,255), 120, 407, 0, "100");
            al_draw_text(displayFuel, al_map_rgb(255,255,255), 290, 243, 0, "200");
            al_draw_text(displayFuel, al_map_rgb(255,255,255), 555, 397, 0, "300");
            al_draw_text(displayFuel, al_map_rgb(255,255,255), 722, 500, 0, "400");
            al_draw_textf(displayFuel, al_map_rgb(255, 255, 225), 20, 20, 0, "Combustible: %d", fuel);
            al_draw_textf(displayVelocity, al_map_rgb(255, 255, 225), 20, 30, 0, "Velocidad: %.2f", v);
            al_draw_textf(displayFuel, al_map_rgb(255,255,255), 780, 20, ALLEGRO_ALIGN_RIGHT, "Presione P para pausa");
            al_draw_bitmap(plataforma1, 0, 240, 0); /*PLATAFORMA 1*/
            al_draw_bitmap(*nave, x , y, 0);
            al_flip_display();
        }
    }

    al_destroy_timer(gametime);
    al_destroy_font(displayFuel);
    al_destroy_font(displayVelocity);
    al_destroy_bitmap(plataforma1);

}


LunarLander/incl/juegoActivo.h
#ifndef JUEGOACTIVO
#define JUEGOACTIVO

#include <allegro5/allegro.h>
#include <allegro5/allegro_audio.h>
#include <allegro5/allegro_acodec.h>
#include <allegro5/allegro_image.h>
#include <allegro5/allegro_font.h>
#include <allegro5/allegro_ttf.h>
#include <stdio.h>

    ALLEGRO_EVENT_QUEUE * queue2;
    ALLEGRO_TIMER * timer2;

    float velocidad(float * v_ini, float * t, float g);
    float gravedad(float g, float * v_ini, float * cont);
    void juego_activo(ALLEGRO_EVENT_QUEUE ** queue, ALLEGRO_EVENT * event, int * onEmpezar, ALLEGRO_BITMAP ** nave, ALLEGRO_BITMAP ** empezarbackground, ALLEGRO_KEYBOARD_STATE * keystate);
    void juego_activo2(ALLEGRO_EVENT_QUEUE ** queue, ALLEGRO_EVENT * event, int * onEmpezar, ALLEGRO_BITMAP ** nave, ALLEGRO_BITMAP ** empezarbackground, ALLEGRO_KEYBOARD_STATE * keystate, int * fuel, int * score);
    void juego_activo3(ALLEGRO_EVENT_QUEUE ** queue, ALLEGRO_EVENT * event, int * onEmpezar, ALLEGRO_BITMAP ** nave, ALLEGRO_BITMAP ** empezarbackground, ALLEGRO_KEYBOARD_STATE * keystate, int * fuel, int * score);
    int pausa(ALLEGRO_BITMAP ** empezarbackground, ALLEGRO_EVENT_QUEUE ** queue, ALLEGRO_EVENT * event, ALLEGRO_KEYBOARD_STATE* keystate);
    void perder(ALLEGRO_BITMAP ** empezarbackground, ALLEGRO_EVENT_QUEUE ** queue, ALLEGRO_EVENT * event, ALLEGRO_KEYBOARD_STATE* keystate);
    void ganar(ALLEGRO_BITMAP ** empezarbackground, ALLEGRO_EVENT_QUEUE ** queue, ALLEGRO_EVENT * event, ALLEGRO_KEYBOARD_STATE* keystate, int * fuel, int * score);

#endif

LunarLander/src/juegoActivo2.c
#include "juegoActivo.h"

void juego_activo2(ALLEGRO_EVENT_QUEUE ** queue, ALLEGRO_EVENT * event, int * onEmpezar, ALLEGRO_BITMAP ** nave, ALLEGRO_BITMAP ** empezarbackground, ALLEGRO_KEYBOARD_STATE * keystate, int * fuel, int * score)
{
    ALLEGRO_TIMER * gametime;
    ALLEGRO_FONT * displayFuel;
    ALLEGRO_FONT * displayVelocity;
    ALLEGRO_BITMAP * plataforma2 = NULL;
    ALLEGRO_FONT * displayPausa;

    displayFuel = al_load_font("resources/fonts/AbyssinicaSIL-R.ttf", 12, 0);
    displayPausa = al_load_font("resources/fonts/AbyssinicaSIL-R.ttf", 12, 0);
    displayVelocity = al_load_font("resources/fonts/AbyssinicaSIL-R.ttf", 12, 0);

    plataforma2 = al_load_bitmap("resources/background/plataforma2.png");


    gametime = al_create_timer(1.0 / 60.0);


    *onEmpezar = 1;
    float x = 10.0;
    float y = 30.0;
    float v = 0;
    float cont = 0.0;
    float cont2 = 0.0;

    al_start_timer(gametime);
    while(*onEmpezar == 1){
        al_wait_for_event(*queue, event);
        al_get_keyboard_state(keystate);
        al_draw_bitmap(*empezarbackground, 0 , 0, 0);

        if(al_key_down(keystate, ALLEGRO_KEY_RIGHT)){
            x += 0.8;
            *fuel -= 2;
        }
        if(al_key_down(keystate, ALLEGRO_KEY_LEFT)){
            x -= 0.8;
            *fuel -= 2;
        }

        if(!(al_key_down(keystate, ALLEGRO_KEY_SPACE))){
            cont = cont + (1.0/60.0);
            v = velocidad(&v, &cont, 0.2);
            y = y + gravedad(0.1, &v, &cont);
            cont2 = (0.4)*cont;
        }else if((al_key_down(keystate, ALLEGRO_KEY_SPACE))){
            cont2 = cont2 + (1.0/60.0);
            v = velocidad(&v, &cont2, -0.3);
            y = y + gravedad(-0.15, &v, &cont2);
            cont = (0.4)*cont2;
            *fuel -= 6;
        }

        if(al_key_down(keystate, ALLEGRO_KEY_P)){
            *onEmpezar = pausa(&*empezarbackground, &*queue, &*event, &*keystate);
        }

        if(*fuel < 0){
            *onEmpezar = 0;
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }

        if(event->type == ALLEGRO_EVENT_DISPLAY_CLOSE){
            *onEmpezar = 0;
        }

        if (y > 427 && x < 19 && x > 0){
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 354 && x > 19 && x < 71) {
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 354 && x > 117 && x < 127) {
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 464 && x > 127 && x < 171){
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 374 && x > 171 && x < 248){
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 441 && x > 248 && x < 260){
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 380 && x > 304 && x < 346){
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 414 && x > 346 && x < 365){
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 388 && x > 365 && x < 409){
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 388 && x > 459 && x < 464){
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > (0.827*x)+20.0 && x > 464 && x < 540){
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 393 && x > 540 && x < 635){
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 457 && x > 635 && x < 709){
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 457 && x > 757 && x < 800){
            *onEmpezar = 0;/*PIERDE*/
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        /*GANADOREEEEEES*/
        if (y > 350 && x > 71 && x < 117 && v < 0.6){
            *score = *score + 100;
	        *onEmpezar = 0;/*GANA*/
            juego_activo3(&*queue, &*event, &*onEmpezar, &*nave, &*empezarbackground, &*keystate, &*fuel, &*score);
	    
        }
        if (y > 440 && x > 260 && x < 304 && v < 0.6){
            *score = *score + 400;
	        *onEmpezar = 0;/*GANA*/
            juego_activo3(&*queue, &*event, &*onEmpezar, &*nave, &*empezarbackground, &*keystate, &*fuel, &*score);
        }
        if (y > 389 && x > 409 && x < 459 && v < 0.6){
            *score = *score + 200;
	        *onEmpezar = 0;/*GANA*/
            juego_activo3(&*queue, &*event, &*onEmpezar, &*nave, &*empezarbackground, &*keystate, &*fuel, &*score);
	    
        }
        if (y > 456 && x > 709 && x < 757 && v < 0.6){
            *score = *score + 300;
	        *onEmpezar = 0;/*GANA*/
            juego_activo3(&*queue, &*event, &*onEmpezar, &*nave, &*empezarbackground, &*keystate, &*fuel, &*score);
        }

        //------------Contrario a ganar---------------------------------------------------

        if (y > 350 && x > 71 && x < 117 && v >= 0.6){
	        *onEmpezar = 0;
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 440 && x > 260 && x < 304 && v >= 0.6){
	        *onEmpezar = 0;
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 389 && x > 409 && x < 459 && v >= 0.6){
	        *onEmpezar = 0;
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }
        if (y > 456 && x > 709 && x < 757 && v >= 0.6){
	        *onEmpezar = 0;
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }

        //si sale de pantalla de juego
        if(x < 0 || x > 765){
            *onEmpezar = 0;
            perder(&*empezarbackground, &*queue, &*event, &*keystate);
        }

        if(event->type == ALLEGRO_EVENT_TIMER){
            al_draw_text(displayFuel, al_map_rgb(255,255,255), 80, 355, 0, "100");
            al_draw_text(displayFuel, al_map_rgb(255,255,255), 410, 394, 0, "200");
            al_draw_text(displayFuel, al_map_rgb(255,255,255), 720, 461, 0, "300");
            al_draw_text(displayFuel, al_map_rgb(255,255,255), 270, 445, 0, "400");
            al_draw_textf(displayFuel, al_map_rgb(255, 255, 225), 20, 20, 0, "Combustible: %d", *fuel);
            al_draw_textf(displayVelocity, al_map_rgb(255, 255, 225), 20, 30, 0, "Velocidad: %.2f", v);
            al_draw_textf(displayPausa, al_map_rgb(255,255,255), 780, 20, ALLEGRO_ALIGN_RIGHT, "Presione P para pausa");
            al_draw_bitmap(plataforma2, 0, 333, 0); /*PLATAFORMA 2*/
            al_draw_bitmap(*nave, x , y, 0);
            al_flip_display();
        }
    }

    al_destroy_timer(gametime);
    al_destroy_font(displayFuel);
    al_destroy_font(displayVelocity);
    al_destroy_font(displayPausa);
    al_destroy_bitmap(plataforma2);
}